src/fu-util.c

```diff
@@ -2393,6 +2551,60 @@ fu_util_security (FuUtilPrivate *priv, gchar **values, GError **error)
 		return FALSE;
 	str = fu_util_security_attrs_to_string (attrs);
 	g_print ("%s\n", str);
+
+	/* can we find a remote with a security attr */
+	remote = fu_util_get_remote_with_security_report_uri (priv, &error_local);
+	if (remote == NULL) {
+		g_debug ("failed to find suitable remote: %s", error_local->message);
+		return TRUE;
+	}
+
+	/* ask if we can upload them to the LVFS */
+	if (priv->no_unreported_check)
+		return TRUE;
+	if (!priv->assume_yes ||
+	    !fwupd_remote_get_automatic_security_reports (remote)) {
+		g_autofree gchar *tmp = NULL;
+		/* TRANSLATORS: ask the user to share, %s is something like:
+		 * "Linux Vendor Firmware Service" */
+		tmp = g_strdup_printf ("Upload these anonymous results to the %s to help other users?",
+				       fwupd_remote_get_title (remote));
+
+		g_print ("\n%s [y|N]: ", tmp);
+		if (!fu_util_prompt_for_boolean (FALSE)) {
+			g_print ("%s [Y|n]: ",
+				 /* TRANSLATORS: stop nagging the user */
+				 _("Ask again next time?"));
+			if (!fu_util_prompt_for_boolean (TRUE)) {
+				if (!fwupd_client_modify_remote (priv->client,
+								 fwupd_remote_get_id (remote),
+								 "SecurityReportURI", "",
+								 NULL, error))
+					return FALSE;
+			}
+			return TRUE;
+		}
+	}
+
+	/* upload, with confirmation */
+	if (!fu_util_upload_security (priv, attrs, remote, error))
+		return FALSE;
+
+	/* as this worked, ask if the user want to do this every time */
+	if (!fwupd_remote_get_automatic_security_reports (remote)) {
+		g_print ("%s [y|N]: ",
+			 /* TRANSLATORS: can we JFDI? */
+			 _("Automatically upload every time?"));
+		if (fu_util_prompt_for_boolean (FALSE)) {
+			if (!fwupd_client_modify_remote (priv->client,
+							 fwupd_remote_get_id (remote),
+							 "AutomaticSecurityReports", "true",
+							 NULL, error))
+				return FALSE;
+		}
+	}
```

Comment:
I think this whole block of code about uploading and the handling of it should go into the static `fu_util_upload_security` function.  Put all the logic around that upload there rather than disjointed looking up a remote here and then checking it it works and then looking it up again there.  Just do it all in one place.

Reply:
Agreed, fixed.